spring-web/src/main/java/org/springframework/web/client/DefaultResponseErrorHandler.java

```diff
@@ -109,20 +111,40 @@ protected boolean hasError(int unknownStatusCode) {
 	 * {@link HttpStatus} enum range.
 	 * </ul>
 	 * @throws UnknownHttpStatusCodeException in case of an unresolvable status code
-	 * @see #handleError(ClientHttpResponse, HttpStatus)
+	 * @see #handleError(URI, HttpMethod, ClientHttpResponse, HttpStatus)
 	 */
 	@Override
 	public void handleError(ClientHttpResponse response) throws IOException {
+		handleError(null, null, response);
+	}
+
+	/**
+	 * Handle the error in the given response with the given resolved status code.
+	 * <p>The default implementation throws:
+	 * <ul>
+	 * <li>{@link HttpClientErrorException} if the status code is in the 4xx
+	 * series, or one of its sub-classes such as
+	 * {@link HttpClientErrorException.BadRequest} and others.
+	 * <li>{@link HttpServerErrorException} if the status code is in the 5xx
+	 * series, or one of its sub-classes such as
+	 * {@link HttpServerErrorException.InternalServerError} and others.
+	 * <li>{@link UnknownHttpStatusCodeException} for error status codes not in the
+	 * {@link HttpStatus} enum range.
+	 * </ul>
+	 * @throws UnknownHttpStatusCodeException in case of an unresolvable status code
+	 * @see #handleError(URI, HttpMethod, ClientHttpResponse, HttpStatus)
+	 */
+	@Override
+	public void handleError(URI url, HttpMethod method, ClientHttpResponse response) throws IOException {
 		HttpStatus statusCode = HttpStatus.resolve(response.getRawStatusCode());
 		if (statusCode == null) {
 			byte[] body = getResponseBody(response);
-			String message = getErrorMessage(response.getRawStatusCode(),
-					response.getStatusText(), body, getCharset(response));
+			String message = getErrorMessage(response.getRawStatusCode(), response.getStatusText(), body, getCharset(response), url, method);
 			throw new UnknownHttpStatusCodeException(message,
 					response.getRawStatusCode(), response.getStatusText(),
 					response.getHeaders(), body, getCharset(response));
 		}
-		handleError(response, statusCode);
+		handleError(url, method, response, statusCode);
```

Comment:
This is the protected method that's now bypassed.

Reply:
Thanks for the explanation!

Then it can definitely wait for a future Spring release. I'm not in a particular hurry - any custom extensions would well enough.

Should I close the PR, or just keep it here? 

Also - is it something that you'd see easy to port into WebClient? Would it make sense to make these two in sync?

Reply:
I think we can close it for now. On the WebClient side it looks we do have the URL and HTTP method in `WebClientResponseException`.

Reply:
Thanks for the feedback @rstoyanchev ! 

Is there a chance to squeeze this in a future Spring Framework release? Should I come back once it's announced?

Reply:
Yes, of course. We can probably re-open this PR then.